posthog/api/person.py

```diff
@@ -237,14 +238,28 @@ def list(self, request: request.Request, *args: Any, **kwargs: Any) -> response.
         elif not filter.limit:
             filter = filter.shallow_clone({LIMIT: DEFAULT_PAGE_LIMIT})
 
-        query, params = PersonQuery(filter, team.pk).get_query(paginate=True, filter_future_persons=True)
-
-        raw_result = insight_sync_execute(
-            query, {**params, **filter.hogql_context.values}, filter=filter, query_type="person_list", team_id=team.pk
+        person_query = PersonQuery(filter, team.pk)
+        paginated_query, paginated_params = person_query.get_query(paginate=True, filter_future_persons=True)
+        raw_paginated_result = insight_sync_execute(
+            paginated_query,
+            {**paginated_params, **filter.hogql_context.values},
+            filter=filter,
+            query_type="person_list",
+            team_id=team.pk,
         )
-
-        actor_ids = [row[0] for row in raw_result]
-        actors, serialized_actors = get_people(team, actor_ids)
+        actor_ids = [row[0] for row in raw_paginated_result]
+        _, serialized_actors = get_people(team, actor_ids)
+
+        total_query, total_params = person_query.get_query(paginate=False, filter_future_persons=True)
+        total_query_aggregated = f"SELECT count() FROM ({total_query})"
```

Comment:
I'd think this query would be slow for projects with a lot of persons (it'd need to check all active parts?) - did you try running this in production or have better knowledge than me and know this is quick?

Reply:
I ran it for out project but indeed for some larger ones it takes a few seconds, which is not ideal… I think when we fully release this the solution will have to be we'll only load the total count once, but for now just put this behind a URL param that's only used in PostHog 3000.